lib/node_modules/@stdlib/ndarray/base/unary-reduce-subarray/lib/3d_blocked.js

```diff
@@ -0,0 +1,226 @@
+*     'order': 'row-major'
+* };
+*
+* // Create an output ndarray-like object:
+* var y = {
+*     'dtype': 'generic',
+*     'data': ybuf,
+*     'shape': ysh,
+*     'strides': sy,
+*     'offset': oy,
+*     'order': 'row-major'
+* };
+*
+* // Initialize ndarray-like objects representing sub-array views:
+* var views = [
+*     {
+*         'dtype': x.dtype,
+*         'data': x.data,
+*         'shape': [ 2, 2 ],
+*         'strides': [ 2, 1 ],
+*         'offset': x.offset,
+*         'order': x.order
+*     }
+* ];
+*
+* // Perform a reduction:
+* blockedunary3d( base, [ x, y ], views, [ 12, 12, 4 ], {} );
+*
+* var arr = ndarray2array( y.data, y.shape, y.strides, y.offset, y.order );
+* // returns [ [ [ true, false, true ] ] ]
+*/
+function blockedunary3d( fcn, arrays, views, strides, opts ) {
+	var bsize;
+	var ybuf;
+	var dv0;
+	var dv1;
+	var dv2;
+	var ov1;
+	var ov2;
+	var sh;
+	var s0;
+	var s1;
+	var s2;
+	var sv;
+	var ov;
+	var iv;
+	var i0;
+	var i1;
+	var i2;
+	var j0;
+	var j1;
+	var j2;
+	var N;
+	var x;
+	var y;
+	var o;
+	var k;
+
+	// Note on variable naming convention: S#, dv#, i#, j# where # corresponds to the loop number, with `0` being the innermost loop...
+
+	N = arrays.length;
+	x = arrays[ 0 ];
+	y = arrays[ 1 ];
+
+	// Resolve the loop interchange order:
+	o = loopOrder( y.shape, strides, y.strides );
+	sh = o.sh;
+	sv = [ o.sx, o.sy ];
+	for ( k = 2; k < N; k++ ) {
+		sv.push( takeIndexed( arrays[k].strides, o.idx ) );
+	}
+	// Determine the block size:
+	bsize = blockSize( x.dtype, y.dtype );
+
+	// Resolve a list of pointers to the first indexed elements in the respective ndarrays:
+	ov = offsets( arrays );
+
+	// Cache a reference to the output ndarray buffer:
+	ybuf = y.data;
+
+	// Cache offset increments for the innermost loop...
+	dv0 = [];
+	for ( k = 0; k < N; k++ ) {
+		dv0.push( sv[k][0] );
+	}
+	// Initialize loop variables...
+	ov1 = zeros( N );
+	ov2 = zeros( N );
+	dv1 = zeros( N );
+	dv2 = zeros( N );
+	iv = zeros( N );
+
+	// Iterate over blocks...
+	for ( j2 = sh[2]; j2 > 0; ) {
+		if ( j2 < bsize ) {
+			s2 = j2;
+			j2 = 0;
+		} else {
+			s2 = bsize;
+			j2 -= bsize;
+		}
+		for ( k = 0; k < N; k++ ) {
+			ov2[ k ] = ov[k] + ( j2*sv[k][2] );
+		}
+		for ( j1 = sh[1]; j1 > 0; ) {
+			if ( j1 < bsize ) {
+				s1 = j1;
+				j1 = 0;
+			} else {
+				s1 = bsize;
+				j1 -= bsize;
+			}
+			// Compute index offsets and loop offset increments for the first ndarray elements in the current block...
+			for ( k = 0; k < N; k++ ) {
+				ov1[ k ] = ov2[k] + ( j1*sv[k][1] );
+				dv1[ k ] = sv[k][1] - ( s1*sv[k][0] );
+			}
+			for ( j0 = sh[0]; j0 > 0; ) {
+				if ( j0 < bsize ) {
+					s0 = j0;
+					j0 = 0;
+				} else {
+					s0 = bsize;
+					j0 -= bsize;
+				}
+				// Compute index offsets and loop offset increments for the first ndarray elements in the current block...
+				for ( k = 0; k < N; k++ ) {
+					iv[ k ] = ov1[k] + ( j0*sv[k][0] );
+					dv2[ k ] = sv[k][2] - ( s0*sv[k][0] );
```

Comment:
Same thing. This logic is off. I suggest using a pencil and paper for how the logic in `ndarray/base/unary` works and why these increments are computed.